src/ORM/Query.php

```diff
@@ -1059,12 +1059,20 @@ protected function _transformQuery()
             return;
         }
 
+        $joins = $this->_parts['join'];
+        $this->_parts['join'] = [];
+
         if (empty($this->_parts['from'])) {
             $this->from([$this->_repository->getAlias() => $this->_repository->table()]);
         }
         $this->_addDefaultFields();
         $this->getEagerLoader()->attachAssociations($this, $this->_repository, !$this->_hasFields);
         $this->_addDefaultSelectTypes();
+
+        $i = count($this->_parts['join']);
+        foreach ($joins as $a => $j) {
+            $this->_parts['join'][is_string($a) ? $a : $i++] = $j;
```

Comment:
Couldn't this result in joins created by the eagerloader being overwritten?

Reply:
The problem is in the order in which JOINS are created.

The eagerloader is processed last.
Indepedent order to call the method.

Reply:
The problem is in the order in which JOINS are created.

The eagerloader is processed last.
Indepedent order to call the method.